src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -481,9 +481,9 @@ const WorkflowView = _.flow(
       // Dockstore users who target floating tags can change their WDL via Github without explicitly selecting a new version in Terra.
       // Before letting the user edit the config we retrieved from the DB, drop any keys that are no longer valid. [WA-291]
       // N.B. this causes `config` and `modifiedConfig` to be unequal, so we (accurately) prompt the user to save before launching
-      const modifiedConfig = filterConfigIO(inputsOutputs)(
-        readSelection ? _.set('rootEntityType', selection.entityType, config) : config
-      )
+      // DO NOT filter when a config is redacted, when there's no IO from the WDL we would erase the user's inputs
+      const unfilteredConfig = readSelection ? _.set('rootEntityType', selection.entityType, config) : config
+      const modifiedConfig = isRedacted ? unfilteredConfig : filterConfigIO(inputsOutputs)(unfilteredConfig)
```

Comment:
Since this is expressing a pipeline of transformations, we could use `_.flow` if that seems clearer.
```suggestion
      const modifiedConfig = _.flow(
        readSelection ? _.set('rootEntityType', selection.entityType) : _.identity,
        !isRedacted ? filterConfigIO(inputsOutputs) : _.identity
      )(config)
```

Reply:
Cool, I will test that out!